test/loco/integer/sendmoremoney2.clj

```diff
@@ -0,0 +1,80 @@
+(ns ^{:author "Andrea Richiardi"
+      :doc "Send + More = Money example + tests, second version." }
+  loco.integer.sendmoremoney2
+  (:refer-clojure :exclude [send]) 
+  (:require [clojure.test :refer :all]
+            [loco.core :refer :all]
+            [loco.constraints :refer :all]
+            [criterium.core :refer [bench]]))
+
+(defn costraint-0-9 [vars]
+  (for [v vars]
+    ($in v 0 9)))
+
+(def letters
+  [:S :E :N :D :M :O :R :Y])
+
+(def carries
+  (for [i (range 4)]
+    ($in [:_c i] 0 1)))
+
+(def ^{:doc "Naive model, $distinct on each letter. Constraint on the sum with carries:
+               c3 c2 c1 c0     
+                S  E  N  D +
+                M  O  R  E =
+             ---------------
+             M  O  N  E  Y  "}
+  smm-naive-model
```

Comment:
You just need a $distinct I guess, same as above, right? I will copy-paste the rest ;)